source/components/JsxParser.tsx

```diff
@@ -260,9 +260,10 @@ export default class JsxParser extends React.Component<TProps> {
 				children = undefined
 			} else if (children.length === 1) {
 				[children] = children
-			} else if (children.length > 1) {
-				// Add `key` to any child that is a react element (by checking if it has `.type`)
-				children = children.map((child, key) => ((child && child.type) ? { ...child, key } : child))
+			} else if (children.length > 1 && !this.props.disableKeyGeneration) {
+				// Add `key` to any child that is a react element (by checking if it has `.type`) if one
+				// does not already exist.
+				children = children.map((child, key) => ((child && child.type) ? { ...child, key: child.key || key } : child))
```

Comment:
This makes sense, though I think it might be a little cleaner as:
```ts
children = children.map((child, key) => (child?.type && !child?.key) ? { ...child, key } : child)
```

This prevents the minor foible of re-spreading the child in the event where it _does_ have a `key` already. What do you think?

Reply:
If you don't mind making the minor change, I'll get this released shortly. Sorry for the long delay.